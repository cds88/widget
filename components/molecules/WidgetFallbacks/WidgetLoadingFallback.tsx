import { CircularProgress, CircularProgressProps } from '@mui/material';
import React from 'react';
import { SPINNER_DATA_TESTIDS } from './const.test';

type WidgetLoadingFallbackProps = CircularProgressProps;

export const WidgetLoadingFallback: React.FC<WidgetLoadingFallbackProps> = (
  props,
) => {
  return (
      <CircularProgress
        aria-busy="true"
        role="progressbar"
        {...props}
        data-testid={SPINNER_DATA_TESTIDS.SPINNER_CIRCULAR_PROGRESS}
      />
  );
};
