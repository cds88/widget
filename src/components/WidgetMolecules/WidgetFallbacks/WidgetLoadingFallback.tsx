import { CircularProgress, CircularProgressProps } from '@mui/material';
import React from 'react';

type WidgetLoadingFallbackProps = CircularProgressProps;

export const WidgetLoadingFallback: React.FC<WidgetLoadingFallbackProps> = (
  props,
) => {
  return (
    <div>
      ABCDEFGHAXYZ
      <CircularProgress
        aria-busy="true"
        role="progressbar"
        {...props}
        data-testid={'loading-spinner'}
      />
    </div>
  );
};
