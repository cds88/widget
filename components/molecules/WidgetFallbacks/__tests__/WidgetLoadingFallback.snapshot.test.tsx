import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { WidgetLoadingFallback } from '../WidgetLoadingFallback'; 

describe('WidgetLoadingFallback Snapshot Tests', () => {
  it('renders correctly with default props', () => {
    const { asFragment } = render(<WidgetLoadingFallback />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom size and color', () => {
    const { asFragment } = render(
      <WidgetLoadingFallback size={60} color="secondary" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
