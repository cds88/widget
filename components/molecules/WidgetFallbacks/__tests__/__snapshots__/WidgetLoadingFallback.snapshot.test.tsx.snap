// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetLoadingFallback Snapshot Tests renders correctly with default props 1`] = `
<DocumentFragment>
  <span
    aria-busy="true"
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
    data-testid="SPINNER_CIRCULAR_PROGRESS"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`WidgetLoadingFallback Snapshot Tests renders with custom size and color 1`] = `
<DocumentFragment>
  <span
    aria-busy="true"
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-w58ak0-MuiCircularProgress-root"
    data-testid="SPINNER_CIRCULAR_PROGRESS"
    role="progressbar"
    style="width: 60px; height: 60px;"
  >
    <svg
      class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </span>
</DocumentFragment>
`;
